src/hooks/useCurrentExecutionStatus.ts

```diff
@@ -0,0 +1,142 @@
+  const { api: targetApi } = useApiTargetPromiseContext();
+  const { sourceChain, targetChain } = useSourceTarget();
+  const laneId = useLaneId();
+  const areApiLoading = useLoadingApi();
+  const { currentTransaction } = useTransactionContext();
+  const {
+    bestBlockFinalized,
+    outboundLanes: { latestReceivedNonce: latestReceivedNonceOnSource }
+  } = useDashboard(SOURCE);
+  const { importedHeaders, bestBlockFinalized: bestBlockFinalizedOnTarget } = useDashboard(TARGET);
+  const { latestReceivedNonceMethodName } = getSubstrateDynamicNames(sourceChain);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+    const getNonceByHash = async () => {
+      const blockHash = await targetApi.rpc.chain.getBlockHash(bestBlockFinalizedOnTarget);
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(
+        latestReceivedNonceMethodName,
+        laneId,
+        blockHash.toJSON()
+      );
+
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      const latestReceivedNonce = latestReceivedNonceCallType.toString();
+      setNonceOfTargetFinalizedBlock(parseInt(latestReceivedNonce));
+    };
+
+    const getLatestReceivedNonce = async () => {
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(latestReceivedNonceMethodName, laneId);
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      setLatestReceivedNonceRuntimeApi(parseInt(latestReceivedNonceCallType.toString()));
+    };
+
+    getNonceByHash();
+    getLatestReceivedNonce();
+  }, [
+    areApiLoading,
+    bestBlockFinalizedOnTarget,
+    currentTransaction,
+    laneId,
+    latestReceivedNonceMethodName,
+    targetApi.registry,
+    targetApi.rpc.chain,
+    targetApi.rpc.state
+  ]);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: number | null) => {
+      if (!transactionValue || !chainValue) {
+        return false;
+      }
+      return chainValue > transactionValue;
+    };
+
+    const isDone = (status: boolean | boolean) => (currentTransaction.block && status ? 'DONE' : 'RUNNING');
+    const step1 = currentTransaction.block ? `included in block ${currentTransaction.block}` : isDone(false);
+    const step2 = stepEvaluator(currentTransaction.block, parseInt(bestBlockFinalized));
+    const step3 = stepEvaluator(currentTransaction.block, parseInt(importedHeaders));
+    const step4 = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceRuntimeApi);
+    const step5 = stepEvaluator(currentTransaction.messageNonce, nonceOfTargetFinalizedBlock);
+    const step6 = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceOnSource);
```

Comment:
Calling the variables `step*` is not super informative. Like it's really hard right now to figure out what the step actually is without external context, I'd be more in favour in naming them properly:

```suggestion
    const sourceTransactionIncluded = currentTransaction.block ? `included in block ${currentTransaction.block}` : isDone(false);
    const sourceTransactionFinalized = stepEvaluator(currentTransaction.block, parseInt(bestBlockFinalized));
    const blockFinalityRelayed = stepEvaluator(currentTransaction.block, parseInt(importedHeaders));
    const messageDelivered = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceRuntimeApi);
    const messageFinalizedOnTarget = stepEvaluator(currentTransaction.messageNonce, nonceOfTargetFinalizedBlock);
    const sourceConfirmationReceived = stepEvaluator(currentTransaction.messageNonce, latestReceivedNonceOnSource);
```